src/main/main.ts

```diff
@@ -341,23 +346,83 @@ const checkPythonEnv = async (splashWindow: BrowserWindowWithSafeIpc) => {
     ipcMain.handle('get-python', () => pythonInfo);
 };
 
-const checkPythonDeps = async (splashWindow: BrowserWindowWithSafeIpc) => {
+const checkSemverGt = (v1: string, v2: string) => {
+    try {
+        return semver.gt(semver.coerce(v1)!.version, semver.coerce(v2)!.version);
+    } catch (error) {
+        log.error(error);
+        return false;
+    }
+};
+
+const checkPythonDeps = async (splashWindow: BrowserWindowWithSafeIpc, hasNvidia: boolean) => {
     log.info('Attempting to check Python deps...');
     try {
         const pipList = await runPipList();
         const installedPackages = new Set(Object.keys(pipList));
 
-        const pending = requiredDependencies.filter((dep) => {
-            return dep.packages.some((pkg) => {
-                if (installedPackages.has(pkg.packageName)) return false;
-                log.info(`Dependency ${dep.name} (${pkg.packageName}) not found.`);
-                return true;
-            });
+        let requiredIndividualPackages: PyPiPackage[] = [];
+        requiredDependencies.forEach((dependency) => {
+            requiredIndividualPackages = [...requiredIndividualPackages, ...dependency.packages];
+        });
```

Comment:
```suggestion
        const requiredIndividualPackages = requiredDependencies.flatMap((dep) => dep.packages);
```

Same for `optionalInvividualPackages` below.